pkg/liquidity-mining/test/GaugeWorkingBalanceHelper.test.ts

```diff
@@ -0,0 +1,339 @@
+        expect(projectedBalanceAfter).to.be.lt(projecteBalanceBefore);
+        expect(projectedRatioAfter).to.be.lt(projectedRatioBefore);
+      });
+    }
+
+    describe('with no veBAL', () => {
+      context('check raw balances', () => {
+        sharedBeforeEach('deposit', async () => {
+          depositIntoGauge(user, stakeAmount);
+        });
+
+        currentBalanceEqualsProjected();
+      });
+
+      context('check balance ratios', () => {
+        sharedBeforeEach('deposit', async () => {
+          depositIntoGauge(user, stakeAmount);
+        });
+
+        currentRatioEqualsProjected();
+      });
+    });
+
+    describe('with a veBAL monopoly', () => {
+      sharedBeforeEach('deposit', async () => {
+        await depositIntoGauge(user, stakeAmount);
+        await createLockForUser(user, bptAmount, LOCK_PERIOD);
+      });
+
+      projectedRatioIsMaxedOut();
+
+      itUpdatesAfterCheckpointing();
+
+      currentRatioEqualsProjected();
+    });
+
+    describe('with 1% veBAL', () => {
+      sharedBeforeEach('deposit', async () => {
+        // Another person stakes 100x as much
+        await depositIntoGauge(user, stakeAmount);
+        await depositIntoGauge(other, stakeAmount);
+
+        await createLockForUser(user, bptAmount, LOCK_PERIOD);
+        await createLockForUser(other, bptAmount.mul(100), LOCK_PERIOD);
+      });
+
+      // Should be less than 5%
+      projectedBalanceIsIncreased(5);
+
+      projectedRatioIncreases();
+
+      itUpdatesAfterCheckpointing();
+
+      veBALDecaysOverTime();
+    });
+  }
+
+  context('on L1', () => {
+    const relativeWeightCap = fp(0.1);
+
+    sharedBeforeEach('deploy helper', async () => {
+      await deployHelper(true);
+    });
+
+    itStoresParameters(true);
+
+    sharedBeforeEach('deploy gauge and factory', async () => {
+      const balTokenAdmin = await deploy('MockBalancerTokenAdmin', { args: [vault.address, BAL.address] });
+      const balMinter = await deploy('MainnetBalancerMinter', {
+        args: [balTokenAdmin.address, gaugeController.address],
+      });
+
+      gaugeImplementation = await deploy('LiquidityGaugeV5', {
+        args: [balMinter.address, veDelegationProxy.address, adaptor.address],
+      });
+      gaugeFactory = await deploy('LiquidityGaugeFactory', { args: [gaugeImplementation.address] });
+
+      const tx = await gaugeFactory.create(pool.address, relativeWeightCap);
+      const event = expectEvent.inReceipt(await tx.wait(), 'GaugeCreated');
+
+      gauge = await deployedAt('LiquidityGaugeV5', event.args.gauge);
+    });
+
+    itComputesWorkingBalances();
+  });
+
+  context('on L2', () => {
+    const version = JSON.stringify({
+      name: 'ChildChainGauge',
+      version: '0',
+      deployment: 'test-deployment',
+    });
+
+    let pseudoMinter: Contract;
+
+    sharedBeforeEach('deploy helper', async () => {
+      await deployHelper(false);
+    });
+
+    itStoresParameters(false);
```

Comment:
```suggestion
    const version = JSON.stringify({
      name: 'ChildChainGauge',
      version: '0',
      deployment: 'test-deployment',
    });
    const readTotalSupplyFromVe = false;

    let pseudoMinter: Contract;

    sharedBeforeEach('deploy helper', async () => {
      await deployHelper(readTotalSupplyFromVe);
    });

    itStoresParameters(readTotalSupplyFromVe);
```

Reply:
Well, I had it as readTotalSupplyFromProxy = false;

Slightly confusing either way.